### PR TITLE
ABS-24: Allow for native token with astroport integration

### PR DESCRIPTION
**Pull request by adairrr**
This change allows the cw-staking adapter to receive a native token (and a cw20) as the staking token for astroport. This is important because on neutron they use the ibc-ed version of their cw20 on terra.

It also creates new neutron features for both the cw-staking and dex packages.

### CODE REVIEW

**Comment by adairrr** [comment]
https://linear.app/abstract-sdk/issue/ABS-24/astroport-native-token-in-cw-staking

**Comment by adairrr** [comment]
<https://linear.app/abstract-sdk/issue/ABS-24/astroport-native-token-in-cw-staking>

File: modules/contracts/adapters/dex/Cargo.toml [hunk 1]
```diff
@@ -32,6 +32,7 @@ juno = [
   "abstract-wyndex-adapter/full_integration",
 ]
 terra = ["dep:terraswap", "abstract-astroport-adapter/full_integration"]
+neutron = ["abstract-astroport-adapter/full_integration"]
```
Comment (Buckram123) [inline modules/contracts/adapters/dex/Cargo.toml:35]: :rocket:

File: modules/contracts/adapters/dex/src/exchanges/exchange_resolver.rs [hunk 2]
```diff
@@ -35,15 +47,15 @@ pub(crate) fn resolve_exchange(value: &str) -> Result<&'static dyn DexCommand, D
...
+        #[cfg(any(feature = "terra", feature = "neutron"))]
         ASTROPORT => Ok(&Astroport {}),
         #[cfg(feature = "kujira")]
         KUJIRA => Ok(&Kujira {}),
         _ => Err(DexError::ForeignDex(value.to_owned())),
     }
 }
 
-/// Given a FULL provider nam (e.g. juno>wyndex), returns wether the request is local or over IBC
+/// Given a FULL provider nam (e.g. juno>wyndex), returns whether the request is local or over IBC
```
Comment (Buckram123) [inline modules/contracts/adapters/dex/src/exchanges/exchange_resolver.rs:58]: :rofl:

**Review by Buckram123 — APPROVED** [review]

File: modules/contracts/adapters/dex/src/exchanges/exchange_resolver.rs [hunk 3]
```diff
@@ -4,20 +4,32 @@ use abstract_adapter_utils::identity::is_current_chain;
...
+#[cfg(feature = "kujira")]
 use crate::exchanges::kujira::{Kujira, KUJIRA};
+#[cfg(feature = "terra")]
 use crate::exchanges::terraswap::{Terraswap, TERRASWAP};
+#[cfg(any(feature = "terra", feature = "neutron"))]
 use abstract_astroport_adapter::{dex::Astroport, ASTROPORT};
+#[cfg(feature = "osmosis")]
 use abstract_osmosis_adapter::{dex::Osmosis, OSMOSIS};
+#[cfg(feature = "juno")]
 use abstract_wyndex_adapter::{dex::WynDex, WYNDEX};
```
[the remark this answers is not in the document]
Comment (Buckram123) [reply modules/contracts/adapters/dex/src/exchanges/exchange_resolver.rs:18]: Or actually, isn't it preferred to have just full path, when used? @adairrr

File: modules/contracts/adapters/dex/src/exchanges/exchange_resolver.rs [hunk 4]
```diff
@@ -4,20 +4,32 @@ use abstract_adapter_utils::identity::is_current_chain;
...
 
 pub(crate) fn identify_exchange(value: &str) -> Result<Box<dyn Identify>, DexError> {
     match value {
+        #[cfg(feature = "juno")]
         JUNOSWAP => Ok(Box::<JunoSwap>::default()),
```
Comment (Buckram123) [inline modules/contracts/adapters/dex/src/exchanges/exchange_resolver.rs:23]: Something like that and it should be a little easier to maintain those features
```suggestion
        crate::exchanges::junoswap::JUNOSWAP => Ok(Box::<crate::exchanges::junoswap::JunoSwap>::default()),
```

File: modules/contracts/adapters/dex/src/exchanges/exchange_resolver.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (adairrr) [reply modules/contracts/adapters/dex/src/exchanges/exchange_resolver.rs:18]: Yeah, nice suggestion!

**Pull request merged**